### PR TITLE
Reply feature

### PR DESCRIPTION
**Pull request by caroljli**
* Implemented reply feature
* Replies show up nested under a reply form with a collapsible toggle
* Behind commits for delete/filter comments -- going to have merge conflicts rip.
    * Also, going to have to make delete query for replies as well once everything is merged.

Screenshots:
image
* Displays comments as nested elements under form.
image
* Added new comments and separate comments are displayed in respective threads.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Reply.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public final class Reply {
+
+  private final long id;
+  private final String name;
+  private final String comment;
+  private final String email;
+  private final String date;
+  private final long parentId;
+
+  public Reply(long id, String name, String comment, String email, String date, long parentId) {
```
Comment (aditij113) [inline portfolio/src/main/java/com/google/sps/data/Reply.java:15]: This looks a lot like the Comment class but has an additional field parentId. Do you think it makes sense to have Reply take (1) Comment and (2) parentId?

File: portfolio/src/main/webapp/assets/script.js [hunk 2]
```diff
@@ -144,64 +144,54 @@ function closeModal() {
...
-  const listElem = document.createElement('li');
-  listElem.innerText = text;
-  return listElem;
-}
 
 /**
  * Fetches comments from server and adds to forum page.
  */
 
 function getComments() {
```
Comment (aditij113) [inline portfolio/src/main/webapp/assets/script.js:152]: very nice job with this function!

File: portfolio/src/main/webapp/forum.html [hunk 3]
```diff
@@ -55,4 +55,40 @@ <h2>Post comments, discussion, or thoughts here!</h2>
         </div>
     </body>
 
+    <!-- jQuery functions -->
+    <script type="text/javascript">
+
+      // checks if comment form contains content
```
Comment (aditij113) [inline portfolio/src/main/webapp/forum.html:61]: Can you make this a bit more specific:

"Disable submit button if form is empty."

File: portfolio/src/main/java/com/google/sps/data/Reply.java (hunk 1 above)
Comment (caroljli) [reply portfolio/src/main/java/com/google/sps/data/Reply.java]: fixed it by just passing 0 as parentId for comments

**Comment by caroljli** [comment]
updated fixes -- please take a look again if you can !! @aditij113 :)

File: portfolio/src/main/java/com/google/sps/data/Reply.java (hunk 1 above)
Comment (aditij113) [reply portfolio/src/main/java/com/google/sps/data/Reply.java]: I don't think I see this code updated yet. Could you check that out? Do you mean you initialized Comment to be a Reply with parentId 0? 

Also, see the other comments in this PR as well.

File: portfolio/src/main/webapp/forum.html (hunk 3 above)
Comment (caroljli) [reply portfolio/src/main/webapp/forum.html]: updated this!

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+/** A comment on the forum page. */
+public final class Comment {
+
+  private final long id;
+  private final String name;
+  private final String comment;
+  private final String email;
+  private final String date;
+
+  public Comment(long id, String name, String comment, String email, String date) {
```
[the remark this answers is not in the document]
Comment (caroljli) [reply portfolio/src/main/java/com/google/sps/data/Comment.java]: added a comment with the specified format!

**Comment by caroljli** [comment]
i can see the updates i made -- let me know if you can't see them still @aditij113

**Pull request merged**